src/sys/signal.rs

```diff
@@ -370,6 +372,15 @@ impl SigSet {
         try!(pthread_sigmask(how, Some(self), Some(&mut oldmask)));
         Ok(oldmask)
     }
+
+    /// Suspends execution of the calling thread until one of the signals in the
+    /// signal mask becomes pending, and returns the accepted signal.
+    pub fn wait(&self) -> Result<SigNum> {
+        let mut signum: SigNum = unsafe { mem::uninitialized() };
```

Comment:
I'm not sure if I like the use of `mem::uninitialized()`, but it's using the pattern already prevalent in the file. Given that, I think it's fine, and could be revisited file-wide. Thoughts @fiveop?